phoebe/atmospheres/models.py

```diff
@@ -0,0 +1,506 @@
+
+        return self
+
+    def parse_rules(self, relative_filename):
+        """
+        Provides rules for parsing atmosphere fits files containing data.
+        Only derived classes should implement this method.
+        """
+
+        return NotImplementedError
+
+    def add_axis_node(self, axis_name, axis_node):
+        """
+        Adds a node to the specified axis. This method is used
+        when we want to add another value into the convex hull
+        spun by the current axes. If you do not know why you would
+        use this method, you likely should not use it.
+
+        Arguments
+        ----------
+        * `axis_name` (string): name of the axis
+        * `axis_node` (float): value of the node to be added
+        """
+
+        if axis_name in self.basic_axis_names:
+            new_axes = list(self.basic_axes)
+            axis_index = self.basic_axis_names.index(axis_name)
+            axis = self.basic_axes[axis_index]
+            if axis_node not in axis:
+                axis = np.append(axis, axis_node)
+                axis.sort()
+
+                new_axes[axis_index] = axis
+                self.basic_axes = tuple(new_axes)
+                nodes = np.vstack([getattr(self, name) for name in self.basic_axis_names]).T
+                self.indices = np.empty_like(nodes, dtype=int)
+                for i, basic_axis in enumerate(self.basic_axes):
+                    self.indices[:, i] = np.searchsorted(basic_axis, nodes[:, i])
+        else:
+            raise ValueError(f"Axis name '{axis_name}' not recognized.")
+
+    def limb_treatment(self, intensities):
+        """
+        Define how intensities at the exact limb (mu=0) should be treated. By
+        default, the intensities are linearly extrapolated to mu=0.
+
+        Arguments
+        ----------
+        * `intensities` (array): intensities across all mus
+
+        Returns
+        --------
+        * an array of intensities with the limb treatment applied.
+        """
+
+        intensities[0] = max(1e-12, intensities[1] + (intensities[2]-intensities[1])/(self.mus[2]-self.mus[1])*(self.mus[0]-self.mus[1]))
+        return intensities
+
+
+class WDBlackbodyModelAtmosphere(ModelAtmosphere):
+    """
+    Wilson-Devinney blackbody model atmosphere.
+    """
+
+    name = 'extern_planckint'
+    external = True
+
+
+class WDKurucz93ModelAtmosphere(ModelAtmosphere):
+    """
+    Wilson-Devinney Kurucz 1993 model atmosphere.
+    """
+
+    name = 'extern_atmx'
+    external = True
+
+
+class BlackbodyModelAtmosphere(ModelAtmosphere):
+    """
+    Blackbody model atmosphere.
+
+    The blackbody model atmosphere is a simple model atmosphere that assumes
+    the object is a blackbody. The grid is defined by a single axis,
+    effective temperature (teffs).
+    """
+
+    name = 'blackbody'
+    prefix = 'bb'
+
+    basic_axis_names = ['teffs']
+    teffs = 10**np.linspace(2.5, 5.7, 97)  # this corresponds to the 316K-501187K range.
```

Comment:
Increase the number of points to try and maintain consistency with previous implementation

Reply:
Can we use `np.geomspace` or `np.logspace`?